### DISCUSSION TITLE
Inbuilt code snippets not available via command pallette, `Ctrl-Shift-P`

### DISCUSSION THREAD

**Discussion by TryerGit** [microsoft/vscode-cpptools#8193] [Q&A] [answered]
In trying to create shortcut keys/prefixes for my user-defined snippets, I would like to make sure that it does not clash with any pre-existing shortcuts/prefixes for other possibly inbuilt snippets. Is there a way to know / look through all currently available code snippets in VSCode and their shortcut keys/prefixes?

I tried to Insert Snippet via the command pallette in the hope that it would reveal all available snippets. Unfortunately, it does not list all snippets. See, for instance, below image, where this command pallette does not show the existence of a `for` snippet and yet inside the editor, when I type `for`, or hit `Ctrl-Space` there is an option to add such a snippet.

See imgur animated gif link [here](https://imgur.com/BnupBoo).

The documentation available at https://code.visualstudio.com/docs/editor/userdefinedsnippets

seems to suggest that all snippets available should be listed either on `Ctrl-Shift-P` (command pallette) followed by `Insert Snippet` or `Ctrl-Space` but `Ctrl-Shift-P` command pallette followed by `Insert Snippet` does not list them, in particular, for instance, the `for` snippet in the image above.

I quote from the documentation:

`You can see the available snippets for a language by running the Insert Snippet command in the Command Palette to get a list of the snippets for the language of the current file. However, keep in mind that this list also includes user snippets that you have defined, and any snippets provided by extensions you have installed.`

I am on VSCode 1.60.2 on Ubuntu. I am on C/C++ Version 1.1.0: November 5, 2020

**Comment by sean-mcmanus** [comment, marked as answer]
Our C/C++ extension snippets are available via Ctrl+Space (autocomplete) and not the "Insert Snippet" command. You can try Ctrl+Space in an empty doc to see the snippets (and keywords too):
![image](https://user-images.githubusercontent.com/19859882/135365150-28c889aa-104d-47f9-baca-8e28702dfdb5.png)
